spyder/app/mainwindow.py

```diff
@@ -1141,30 +1141,24 @@ def post_visible_setup(self):
         Actions to be performed only after the main window's `show` method
         is triggered.
         """
-        # Populate `Panes > View` menu.
-        # This **MUST** be done before restoring the last visible plugins, so
-        # that works as expected.
-        self.layouts.create_plugins_menu()
-
-        # Restore last visible plugins.
-        # This **MUST** be done before running on_mainwindow_visible for
-        # plugins so that the user doesn't experience sudden jumps in the
-        # interface.
-        self.layouts.restore_visible_plugins()
+        # This must be run before the window is shown.
```

Comment:
Maybe a reference to the original issue could be worthy here @ccordoba12 ?

Reply:
Good idea. I amended my last commit with it.